src/hooks/useTxDetails/useTxDetails.ts

```diff
@@ -44,7 +49,7 @@ export interface TxDetails {
   ensTo?: string
   from: string
   ensFrom?: string
-  type: chainAdapters.TradeType | chainAdapters.TxType | ''
+  type: chainAdapters.TradeType | chainAdapters.TxType | StakeType | ''
```

Comment:
this gives me the most pause of anything. this should require a new unchained tx parser and be coming from chain adapters.

@0xApotheosis thoughts?

Reply:
@stackedq to confirm - my understanding of this issue / PR is it's meant to be a front-end only ticket, as there is a corresponding `unchained` issue that handles the parser.

@vividwood did reach out to me about this and I said to stick to `web` changes - but thinking about it now, we should probably just move `StakeType` to Chain Adapters now in anticipation of that.

Even better, @vividwood since you're in context do you want to pick up the `unchained` ticket too and see this feature through from end to end? We can increase the bounty amount to cover it all.

Reply:
Hi, sorry for delayed response, haven't seen a notification. My initial thoughts were that we just add new `ContractMethod` for this (like `StakeUniV2` or something) and then assign `StakeType` in `useTxDetails`. If it is not the case, then sure, we can discuss extending `unchained`

Reply:
@0xApotheosis yeah this issue is meant to deal with the front-end only.
@vividwood I believe you got to use `Stake` and `Unstake` keys from `ContractMethod` enum here.

Reply:
Thanks @stackedq. @vividwood do you feel sufficiently unblocked to continue?

Reply:
> Thanks @stackedq. @vividwood do you feel sufficiently unblocked to continue?

Yes, I will continue working on this

Reply:
@stackedq as far as I understand, `Stake` and `Unstake` from `ContractMethod` are for foxy staking only and already exist (as `TransactionSend/TransactionReceive`). What exactly needs to be done?

Reply:
@0xApotheosis since you have way more information on the context here than I do, could you please help @vividwood out here? I guessed `ContractMethod `'s `Stake` and `Unstake` could be used for this matter here, but I'm not sure if they're only for Foxy or nah.